packages/contracts/test/forwarder/forwarder.ts

```diff
@@ -0,0 +1,306 @@
+            await forwarderWrapper.buyTokensAsync([orderWithFees], [feeOrder], fillAmount, takerAddress);
+            const newBalances = await erc20Wrapper.getBalancesAsync();
+            const takerBalanceAfter = newBalances[takerAddress][defaultMakerAssetAddress];
+
+            const acceptPerc = 98;
+            const acceptableThreshold = takerBalanceBefore.plus(fillAmount.times(acceptPerc).dividedBy(100));
+            const withinThreshold = takerBalanceAfter.greaterThanOrEqualTo(acceptableThreshold);
+            expect(withinThreshold).to.be.true();
+            expect(newBalances[forwarderContract.address][weth.address]).to.be.bignumber.equal(new BigNumber(0));
+        });
+    });
+    describe('buyTokensFee', () => {
+        it('should fill the order and send fee to fee recipient', async () => {
+            const initEthBalance = await web3Wrapper.getBalanceInWeiAsync(feeRecipientAddress);
+            const fillAmount = signedOrder.takerAssetAmount.div(2);
+            const feeProportion = 150; // 1.5%
+            const txHash = await forwarderWrapper.buyTokensFeeAsync(
+                [signedOrder],
+                [],
+                fillAmount,
+                feeProportion,
+                feeRecipientAddress,
+                takerAddress,
+            );
+            const newBalances = await erc20Wrapper.getBalancesAsync();
+            const makerBalanceBefore = erc20Balances[makerAddress][defaultMakerAssetAddress];
+            const makerBalanceAfter = newBalances[makerAddress][defaultMakerAssetAddress];
+            const takerBalanceAfter = newBalances[takerAddress][defaultMakerAssetAddress];
+            const afterEthBalance = await web3Wrapper.getBalanceInWeiAsync(feeRecipientAddress);
+            const takerBoughtAmount = takerBalanceAfter.minus(erc20Balances[takerAddress][defaultMakerAssetAddress]);
+
+            expect(makerBalanceAfter).to.be.bignumber.equal(makerBalanceBefore.minus(takerBoughtAmount));
+            expect(afterEthBalance).to.be.bignumber.equal(
+                initEthBalance.plus(fillAmount.times(feeProportion).dividedBy(10000)),
+            );
+            expect(newBalances[forwarderContract.address][weth.address]).to.be.bignumber.equal(new BigNumber(0));
+        });
+        it('should fail if the fee is set too high', async () => {
+            const initEthBalance = await web3Wrapper.getBalanceInWeiAsync(feeRecipientAddress);
+            const fillAmount = signedOrder.takerAssetAmount.div(2);
+            const feeProportion = 1500; // 15.0%
+
+            try {
+                const txHash = await forwarderWrapper.buyTokensFeeAsync(
+                    [signedOrder],
+                    [],
+                    fillAmount,
+                    feeProportion,
+                    feeRecipientAddress,
+                    takerAddress,
+                );
+                expect.fail(); // Never reached
+            } catch (err) {
```

Comment:
What is the error thrown? Anything worth checking?

Reply:
It will be a revert, once ganache-cli supports returning error messages we can check the error message matches